### ISSUE TITLE
sequence tagger in Farsi 

### ISSUE THREAD

**Issue by myeghaneh** [flairNLP/flair#2733]
I want to  use 'farsi embedding'  (    farsi_embedding = WordEmbeddings('fa')
)for sequence tagging, I used these on these data

```
 N B-P
تفکیک N I-P
درست ADV I-P
و CON I-P
همیشه ADV I-P
زباله N I-P
هایمان N I-P
آزار N I-P
دهنده N I-P
و CON I-P
دست N I-P
و CON I-P
پا N I-P
گیر V_PR I-P
است. DELM I-P
سه N B-P
کیسه N I-P
مختلف ADJ I-P

```

I have used 
```
data_folder = '../data/'
columns = {0: 'text', 1: 'pos', 2: 'BIO'}
n=3
kf = KFold(n_splits=n, random_state=1, shuffle=True)
results = []
for train_index, val_index in kf.split(DATA):
  # splitting Dataframe (dataset not included)
    train_df = DATA.iloc[train_index]
    test_df =DATA.iloc[val_index]
    with open('../data/trainCMTPersianVV03.txt', 'w', encoding = "utf-8-sig") as f:
        for l in train_df:
            for tpl in l:
                f.write('{} {} {}'.format(tpl[0],tpl[1], tpl[2]))
                f.write('\n')   
            f.write('\n')
    with open('../data/testCMTPersianVV03.txt', 'w', encoding = "utf-8-sig") as f:
        for l in test_df:
            for tpl in l:
                f.write('{} {} {}'.format(tpl[0],tpl[1], tpl[2]))
                f.write('\n')
            f.write('\n')
    
        # 1. get the corpus
    corpus:Corpus = ColumnCorpus(data_folder, columns,train_file='trainCMTPersianVV03.txt',test_file='testCMTPersianVV03.txt')
    print(corpus)

    tag_type = 'BIO'

    tag_dictionary = corpus.make_tag_dictionary(tag_type=tag_type)
    print(tag_dictionary)



    

      
    
    farsi_embedding = WordEmbeddings('fa')
   


    tagger: SequenceTagger = SequenceTagger(hidden_size=256,
                                            embeddings=farsi_embedding,
                                            tag_dictionary=tag_dictionary,
                                            tag_type=tag_type, 
                                            )


    trainer: ModelTrainer = ModelTrainer(tagger, corpus)

    trainer.train('resources/taggers/example-ParsiV01', 
                  learning_rate=0.1,
                  mini_batch_size=32,
                  max_epochs=150)
    

```
 I do not know why the result stay in zero, however, the loss is decreasing ;

```
2022-04-19 23:10:51,384 epoch 150 - iter 1/1 - loss 29.68092346 - samples/sec: 25.61 - lr: 0.100000
2022-04-19 23:10:51,386 ----------------------------------------------------------------------------------------------------
2022-04-19 23:10:51,387 EPOCH 150 done: loss 29.6809 - lr 0.1000000
2022-04-19 23:10:51,428 DEV : loss 20.60346031188965 - score 0.0
2022-04-19 23:10:51,430 BAD EPOCHS (no improvement): 0
saving best model
2022-04-19 23:10:54,868 ----------------------------------------------------------------------------------------------------
2022-04-19 23:10:54,869 Testing using best model ...
2022-04-19 23:10:54,870 loading file resources\taggers\example-ParsiV01\best-model.pt
2022-04-19 23:10:55,247 0.0000	0.0000	0.0000
2022-04-19 23:10:55,248 
Results:
- F1-score (micro) 0.0000
- F1-score (macro) 0.0000

By class:
C          tp: 0 - fp: 13 - fn: 13 - precision: 0.0000 - recall: 0.0000 - f1-score: 0.0000
P          tp: 0 - fp: 0 - fn: 52 - precision: 0.0000 - recall: 0.0000 - f1-score: 0.0000
2022-04-19 23:10:55,249 --------------------------------------------------------------------------------

```

any idea what is the problem ?

**Comment by helpmefindaname** [comment]
Hello, judging from your loss, I would assume that your hyperparameters are not a good choice for your dataset.
Usually the lerning rate is the most important parameter to tune.

Look at the training loss of the first epoch the loss should be drastically deceasing over the batches. If it is decreasing very slowly, you might have a very low learning rate. If it is increasing, your learning rate is too high.
When you got to a learning rate that fits good, you'll have to look at other parameters (mainly hidden_states and number of layers) and get it to a point where the dev loss is also decreasing.
You can assume that a loss >1 is already bad for both. 

Besides the parameter tuning, I (as someone who doesn't understand the language) am confused about the example tagging. Usually there are a lot of O tags with a few exceptions, eg:
This is a sentence made in Vienna    .
O     O O O            O    O   B-LOC  O

**Comment by myeghaneh** [comment]
thank you for your answer, it is very nice of you. 

I will try to change Hyper-parameter 
start from learning rate and then mainly hidden_states and number of layers)  and of course I will observe  dev loss 


correct, it is a bit complicated, I am working  on Calim(c)- Premise(p) classification  the stages are 

BIO scheme

so B-P  means the beginning of the Premise 
I-P    means    inside of  Premise  

same for Claim (c)


I have done this on english,  I got very good result, but somehow it does not work with Persian, that might be points that you mentioned or embedding is not good. Since the result are zero and loss was decreasing, I had this idea, maybe there is some fundamental problem such as tagging mismatch or embeddings or... 


I am working on hyperparmeter, but if you have any other idea please let me know

**Comment by myeghaneh** [comment]
> Hello, judging from your loss, I would assume that your hyperparameters are not a good choice for your dataset. Usually the lerning rate is the most important parameter to tune.
> 
> Look at the training loss of the first epoch the loss should be drastically deceasing over the batches. If it is decreasing very slowly, you might have a very low learning rate. If it is increasing, your learning rate is too high. When you got to a learning rate that fits good, you'll have to look at other parameters (mainly hidden_states and number of layers) and get it to a point where the dev loss is also decreasing. You can assume that a loss >1 is already bad for both.
> 
> Besides the parameter tuning, I (as someone who doesn't understand the language) am confused about the example tagging. Usually there are a lot of O tags with a few exceptions, eg: This is a sentence made in Vienna . O O O O O O B-LOC O

thank you again for your response, I have tereid with the different learning rate, unfortunately still a bad loss, and zeo, any idea? the things is that that works for original language (englsih) It should works also for Perisna in My view, I dod not know why
